modules/dnn/src/tensorflow/tf_importer.cpp

```diff
@@ -2813,13 +2912,7 @@ void TFImporter::populateNet()
             {
                 if (layout != DATA_LAYOUT_UNKNOWN)
                 {
-                    if (it->second == DATA_LAYOUT_UNKNOWN)
-                        it->second = layout;
-                    else if (it->second != layout)
-                    {
-                        it->second = DATA_LAYOUT_UNKNOWN;
-                        layout = DATA_LAYOUT_UNKNOWN;
-                    }
+                    it->second = layout;
```

Comment:
Is this change necessary? This doesn't seem to affect tests. If this fixes something in your model, I propose to open a new PR, describing the problem.

Reply:
I will create a new PR for that.